rpc/Base.ts

```diff
@@ -123,40 +142,59 @@ export abstract class Client<
     methodName: MethodName,
     params: Parameters<Config_["RpcSubscriptionMethods"][MethodName]>,
     createListenerCb: CreateListenerCb<Config_, MethodName>,
+    createErrorListenerCb: CreateInternalErrorListenerCb<InternalError> = defaultCreateListenerCb,
     cleanup: SubscriptionCleanup<Config_, MethodName> = () => Promise.resolve(),
-  ): Promise<undefined | msg.ErrMessage<Config_>> => {
+  ): Promise<undefined | msg.ErrMessage<Config_> | InternalError> => {
     const initRes = await this.call(methodName, params);
-    if (initRes.error) {
+    if (initRes instanceof Error) {
+      return initRes;
+    } else if (initRes.error) {
       // TODO: fix typings
       return initRes as msg.ErrMessage<Config_>;
     }
     const cleanupApplied = () => cleanup(initRes as msg.OkMessage<Config_, MethodName>);
     const terminalPending = deferred<undefined | msg.ErrMessage<Config_>>();
-    this.listen((stop) => {
-      const listenerCb = createListenerCb(
-        async () => {
-          stop();
-          await cleanupApplied();
-          terminalPending.resolve();
-        },
-      );
-      return (res) => {
-        if (res.params?.subscription && res.params.subscription === initRes.result) {
-          listenerCb(res as msg.NotifMessage<Config_, MethodName>);
-        } else {
-          // TODO: associate errors with subscriptions & exit
-        }
-      };
-    });
+    this.listen(
```

Comment:
This will change depending on whatever final interface we settle on –– will postpone reviewing this portion